src/main/java/Problem2.java

```diff
@@ -1,7 +1,37 @@
 public class Problem2 {
 
+    private static int index = 1;
+    private static int swap = 0;
+
     // Do not change signature (function name, parameters)
     public static void bubbleSort(int[] data) {
         // Add your solution
+        if (data != null && data.length >= 2) {
+            int lhs, rhs;
+
+                lhs = data[index - 1];
+                rhs = data[index];
+
+                if (lhs > rhs) {
+                    data[index - 1] = rhs;
+                    data[index] = lhs;
+                    swap++;
+                }
+
+                //if completely set, also index needs to be reset b/c instance variable
+                if (index == data.length - 1 && swap == 0) {
+                    index = 1;
+                    return;
+                }
+                //if reached end and not fully set
+                else if(index == data.length - 1) {
```

Comment:
I'm still trying to figure out if this is really bubble sort algorithm. never seen it done this way to be honest.

it's usually a bad idea though to add member variable to help recursion to work. 

please rewrite this in a simpler way. no recursion is need for bubble sort. nested for loops would work just fine.